pkg/controller/instance/instance_controller.go

```diff
@@ -549,12 +549,12 @@ func getPlanToBeExecuted(i *v1beta1.Instance, ov *v1beta1.OperatorVersion) (*str
 		plan := kudov1beta1.SelectPlan([]string{v1beta1.CleanupPlanName}, ov)
 		if plan != nil {
 			if planStatus := i.PlanStatus(*plan); planStatus != nil {
-				if !planStatus.Status.IsRunning() {
-					if planStatus.Status.IsFinished() {
-						// we already finished the cleanup plan
-						return nil, nil
-					}
+				switch planStatus.Status {
+				case kudov1beta1.ExecutionNeverRun:
 					return plan, nil
+				case kudov1beta1.ExecutionComplete:
```

Comment:
What happens in the case of `ExecutionFatalError`?

We should probably check for `planStatus.isTerminal`, i guess?

Reply:
I just looked at the surrounding code. I think the `if i.isDeleting() {` block should always return, at the moment it falls through to `i.GetPlanInProgress() != nil {` etc. This might lead to the correct result, but it's not really clear from the code what actually should happen

Reply:
> I think the `if i.isDeleting() {` block should always return, at the moment it falls through to `i.GetPlanInProgress() != nil {`

the `i.GetPlanInProgress` is **outside** of the `i.isDeleting` block. The problem here is/was that `if !planStatus.Status.IsRunning()` on the left side **also covers** the `ExecutionFatalError` case, where the `plan` is returned here and then reseted/restarted in the caller (we obviously don't want to infinitely restart plans in `FATAL_ERROR` status). 

I'm reworking this part of logic as part of my PR so I would go for a bigger refactoring here.

Reply:
Yes, the left side is certainly not correct. I'm just noted that the new code does not clearly states what happens in case of a `ExecutionFatalError`. If we add an additional
```
case kudov1beta1.ExecutionFatalError:
   // Don't restart in case of a fatal error
  return nil, nil
```
it would feel a lot more clear to me.

Reply:
This is implicitly covered by the subsequent `i.GetPlanInProgress()` check and thus redundant but I don't mind.

Reply:
I'd like to see it changed, but it's not really a requirement for me, so approving for now.